src/main/java/seedu/address/ui/PersonCard.java

```diff
@@ -71,24 +61,32 @@ public PersonCard(ReadOnlyPerson person, int displayedIndex) {
 
     /**
      * This method takes in the tagName, and returns the color associated with that tagName
-     * If the if the tag has no associated color, a random one from the list available will be given.
-     * Once the list runs out, all other tags will be grey.
-     *
+     * If the if the tag has no associated color, a unique random color will be assigned to the tag.
+=     *
      * @param tagName is the String name of the tag
      * @return the color associated to the tagName
      */
     public static String obtainTagColors(String tagName) {
         if (!currentTagColors.containsKey(tagName)) {
-            if (availableColorsLeft.size() != 0) {
-                Random rand = new Random();
-                int randIndex = rand.nextInt(availableColorsLeft.size());
-                assignedColor = availableColorsLeft.get(randIndex);
-
-                currentTagColors.put(tagName, assignedColor);
-                availableColorsLeft.remove(randIndex);
-            } else {
-                currentTagColors.put(tagName, "GRAY");
-            }
+            do {
+                Random random = new Random();
+                final float hue = random.nextFloat();
+                final float saturation = 0.65f + random.nextFloat()
+                        * (0.90f - 0.65f);
+                final float luminance = 0.60f + random.nextFloat()
+                        * (0.90f - 0.60f);
+
+                Color color = Color.getHSBColor(hue, saturation, luminance);
+
+                Formatter hexRepresentation = new Formatter(new StringBuffer("#"));
+                hexRepresentation.format("%02X", color.getRed());
+                hexRepresentation.format("%02X", color.getGreen());
+                hexRepresentation.format("%02X", color.getBlue());
+                assignedColor = hexRepresentation.toString();
+            } while (usedColors.contains(assignedColor));
```

Comment:
Since the colours are randomly generated, is it possible that the randomly generated colour may be difficult to read with our current white font?

Reply:
For my color, there is 3 factors (hue, saturation, luminescence) that will affect how my color will look like. I set the random range of the 3 factors such that the tag color will not clash with the font color.

Reply:
You can checkout this branch to see if the colors are appropriate, if not I can change it to a more fitting range of colors.